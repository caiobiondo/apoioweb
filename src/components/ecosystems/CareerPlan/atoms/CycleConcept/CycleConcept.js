import React, { Component } from 'react';
import propTypes from 'prop-types';
import { Tooltip } from 'react-tippy';
import 'react-tippy/dist/tippy.css';

import {
  CycleConceptWrapper,
  CycleConceptValue,
  ConceptTooltipWrapper,
} from './CycleConcept.styles';

export default class CycleConcept extends Component {
  renderTooltipContent() {
    const { concept, size } = this.props;

    return (
      <ConceptTooltipWrapper>
        <CycleConceptValue concept={concept} size={size} /> {concept}
      </ConceptTooltipWrapper>
    );
  }

  render() {
    const { concept, size } = this.props;

    return (
      <CycleConceptWrapper>
        <Tooltip
          disabled={!concept}
          position="bottom"
          className="Tooltip"
          theme="light"
          html={this.renderTooltipContent()}
        >
          <CycleConceptValue concept={concept} size={size} />
        </Tooltip>
      </CycleConceptWrapper>
    );
  }
}

CycleConcept.propTypes = {
  concept: propTypes.string.isRequired,
  size: propTypes.string,
};

CycleConcept.defaultProps = {
  size: 'normal',
};
