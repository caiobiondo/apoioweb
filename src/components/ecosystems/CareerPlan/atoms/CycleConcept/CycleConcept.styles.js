import styled from 'styled-components';
import { gray700 } from 'styles/colors';
import { RobotoRegular } from 'styles/typography';

import { IndicatorConceptsColors } from 'components/ecosystems/CareerPlan/enums/IndicatorConcepts';

export const CycleConceptWrapper = styled.div``;

export const CycleConceptValue = styled.span`
  background-color: white;
  border-radius: 3px;
  border: 1px solid ${gray700};
  display: inline-block;
  height: 7px;
  margin-top: 5px;
  width: 12px;

  ${({ concept }) =>
    concept &&
    `
    background: ${IndicatorConceptsColors[concept]};
    border: none;
  `};

  ${({ size }) =>
    size === 'large' &&
    `
    margin-bottom: 30px;
    height: 15px;
  `};
`;

export const ConceptTooltipWrapper = styled.div`
  padding: 10px 15px;
  font-family: ${RobotoRegular};
  text-transform: uppercase;
  font-size: 14px;
  color: ${gray700};

  ${CycleConceptValue} {
    margin: -3px 10px 0;
    vertical-align: middle;
  }
`;
