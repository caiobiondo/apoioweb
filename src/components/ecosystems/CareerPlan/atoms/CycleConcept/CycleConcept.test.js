import React from 'react';
import { shallow } from 'enzyme';
import CycleConcept from './CycleConcept';

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides);

  const result = shallow(<CycleConcept {...props} />);

  return {
    props,
    result,
  };
};

describe('Cycle concept', () => {
  it('should render the concept label with the correctly props', () => {
    // given

    // when
    const { result } = setup({ concept: 'Excede' });

    // then
    expect(result).toMatchSnapshot();
  });

  it('should disable the tooltip when there is not a concept', () => {
    // given

    // when
    const { result } = setup({ concept: '' });

    // then
    expect(result.find('Tooltip').props().disabled).toBeTruthy();
  });
});
