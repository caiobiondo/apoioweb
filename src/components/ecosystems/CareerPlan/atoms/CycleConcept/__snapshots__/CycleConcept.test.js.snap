// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cycle concept should render the concept label with the correctly props 1`] = `
<CycleConceptstyles__CycleConceptWrapper>
  <Tooltip
    animateFill={true}
    animation="shift"
    arrow={false}
    arrowSize="regular"
    className="Tooltip"
    delay={0}
    disabled={false}
    distance={10}
    duration={375}
    followCursor={false}
    hideDelay={0}
    hideDuration={375}
    hideOnClick={true}
    html={
      <CycleConceptstyles__ConceptTooltipWrapper>
        <CycleConceptstyles__CycleConceptValue
          concept="Excede"
          size="normal"
        />
         
        Excede
      </CycleConceptstyles__ConceptTooltipWrapper>
    }
    inertia={false}
    interactive={false}
    interactiveBorder={2}
    multiple={false}
    offset={0}
    onHidden={[Function]}
    onHide={[Function]}
    onRequestClose={[Function]}
    onShow={[Function]}
    onShown={[Function]}
    popperOptions={Object {}}
    position="bottom"
    size="regular"
    sticky={false}
    stickyDuration={200}
    style={Object {}}
    theme="light"
    touchHold={false}
    trigger="mouseenter focus"
    unmountHTMLWhenHide={false}
  >
    <CycleConceptstyles__CycleConceptValue
      concept="Excede"
      size="normal"
    />
  </Tooltip>
</CycleConceptstyles__CycleConceptWrapper>
`;
